config/samples/resources/gkehubfeaturemembership/config-management-feature-membership/gkehub_v1beta1_gkehubfeaturemembership.yaml

```diff
@@ -20,6 +20,7 @@ spec:
   projectRef:
     name: gkehubfeaturemembership-dep-acm
   location: global
+  membershipLocation: us-central1-a
```

Comment:
What if the referenced GKEHubMembership has a different location compared with this field?
Asking because I'm curious whether this field should actually be a reference field or should not be explicitly configured.

Also to educate myself, is it a common use case to have membershipLocation different from location like this?

Reply:
+1 on this. http://b/291750161#comment4 does not set the field `membershipLocation`.

Reply:
The referenced membership resource spec.location needs to match the spec.membershipLocation for this resource. Otherwise, the dependency won't be found. I am running sample tests locally, the test should verify the behaviour of explicitly setting this field.

location refers to the location of a feature https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/crds/gkehub_v1beta1_gkehubfeaturemembership.yaml#L351. Feature resource is global and membership could be regional, so it is common they have different values.

Reply:
Thanks for the link. Can you add a comment onto the `location` field to clarify that it is for feature?

Reply:
If the referenced membership `resource spec.location` needs to match the `spec.membershipLocation` for this feature membership resource, then I think this should be a reference field, otherwise this implicit dependency/requirement could introduce additional cost to users. Could you file a GH issue about the idea of making this field optional in the long term? The direct controller should be able to read the location from the referenced GKEHubMembership to fill out this field.

Reply:
Yes, I can createhttps://github.com/GoogleCloudPlatform/k8s-config-connector/issues/2553#issue-2478836209